project_generator/project.py

```diff
@@ -0,0 +1,401 @@
+        toolchain_specific_settings = self.tool_specific[tool]
+
+        if tool == self.TOOLCHAINS[tool]:
+            # make sure we don't get duplicates :)
+            tool_specific_settings = ToolSpecificSettings()
+
+        d = {
+            'name': self.name,
+            'mcu': self.mcu,
+            'core': self.core,
+            'target': self.target,
+            'include_paths': self.include_paths + tool_specific_settings.include_paths,
+            'source_paths': self.source_paths + tool_specific_settings.source_paths,
+            'source_files': merge_recursive(self.source_groups, 
+                                            tool_specific_settings.source_groups,
+                                            toolchain_specific_settings.source_groups),
+            # for backwards compatibility
+            'source_files_c': [merge_recursive(self.source_of_type('c'), 
+                                               tool_specific_settings.source_of_type('c'),
+                                               toolchain_specific_settings.source_of_type('c'))],
+            'source_files_cpp': [merge_recursive(self.source_of_type('cpp'),
+                                                 tool_specific_settings.source_of_type('cpp'),
+                                                 toolchain_specific_settings.source_of_type('cpp'))],
+            'source_files_s': [merge_recursive(self.source_of_type('s'), 
+                                               tool_specific_settings.source_of_type('s'),
+                                               toolchain_specific_settings.source_of_type('s'))],
+            'source_files_obj': self.all_sources_of_type('obj') + 
+                                tool_specific_settings.all_sources_of_type('obj') +
+                                toolchain_specific_settings.all_sources_of_type('obj'),
+            'source_files_lib': self.all_sources_of_type('lib') + 
+                                tool_specific_settings.all_sources_of_type('lib') +
+                                toolchain_specific_settings.all_sources_of_type('lib'),
+            'linker_file': tool_specific_settings.linker_file
+                        or toolchain_specific_settings.linker_file
+                        or self.linker_file,
+            'macros': self.macros + 
+                      tool_specific_settings.macros +
+                      toolchain_specific_settings.macros,
+            'misc': [merge_recursive(tool_specific_settings.misc, toolchain_specific_settings.misc)],
+            'project_dir': self.project_dir
+        }
+
+        return d
+
+    @staticmethod
+    def scrape_dir(directory, project_name, board):
+        data = {
+            'common': {
+                'linker_file': [],
+                'source_files': [],
+                'include_paths': [],
+                'target': ''
+            }
+        }
+
+        linker_filetypes = ['sct', 'ld', 'lin']
```

Comment:
@aethaniel thanks for letting me know I have no experience with IAR so I'll fix that at some point